webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/lift/ComplianceApi.scala

```diff
@@ -757,49 +793,68 @@ class ComplianceAPIService(
       compliance   <- getGlobalComplianceMode().toIO
       reports      <- reportingService
                         .findRuleNodeStatusReports(
-                          nodeInfos.keySet,
+                          nodeTmpInfos.keySet,
                           rules.map(_.id).toSet
                         )
                         .toIO
     } yield {
 
-      // get nodeIds by rules
-      val nodeByRules = rules.map(rule => (rule, RoNodeGroupRepository.getNodeIds(allGroups, rule.targets, allNodeInfos)))
+      // A map to access the node fqdn and settings
+      val nodeInfos = nodeTmpInfos.view.mapValues(n => (n.name, n)).toMap
+
+      val nodeAndPolicyModeByRules = rules
```

Comment:
I wonder if we shouldn't have a cache for that. 
The "node/rule policy mode" is something that is costly to compute, and vary only at specific point (new node, change on rule/directive, change on groups - that's all?). So it looks like a good candidate to keep these data. I think actually the changes are the same than when a generation is needed, and the same than what we would like to have in the expected reports. 

So, perhaps not something for that PR, but we need to keep in mind that when perf will need to get better